plugins/inputs/socket_listener/socket_listener_test.go

```diff
@@ -262,6 +262,93 @@ func TestLargeReadBuffer(t *testing.T) {
 	testutil.RequireMetricsEqual(t, expected, actual, testutil.IgnoreTime())
 }
 
+func TestLargeReadBufferUnixgram(t *testing.T) {
+	// Construct a buffer-size setting of 100KiB
+	// Assuming that the testing environment has net.core.wmem_max set to a value greater than 100KiB
+	if runtime.GOOS == "windows" {
+		t.Skip("Skipping on Windows, as unixgram sockets are not supported")
+	}
+
+	if runtime.GOOS == "darwin" {
+		t.Skip("Skipping on macOS (darwin), as unixgram write buffer size cannot be changed (default 2048 bytes)")
+	}
```

Comment:
Can we remove this check as this will be caught by the `unixConn.SetWriteBuffer(len(message))` check below?

Reply:
SetWriteBuffer will not return an error but will set the maximum possible defined by wmem_max. I've tried it.